src/Uno.UI/UI/Xaml/Controls/GridView/GridViewItem.cs

```diff
@@ -17,5 +17,14 @@ public GridViewItem()
 		partial void Initialize();
 
 		public global::Windows.UI.Xaml.Controls.Primitives.GridViewItemTemplateSettings TemplateSettings { get; } = new Primitives.GridViewItemTemplateSettings();
+
+		private protected override void OnLoaded()
+		{
+			base.OnLoaded();
+			if (Selector is GridView gv)
+			{
+				ApplyMultiSelectState(gv.IsSelectionMultiple);
+			}
+		}
```

Comment:
Interesting! It's likely that this particular override can be placed directly in `SelectorItem.cs`. It does not need to be knowing that it's dealing with a `GridView`, and it can update the multi-selection state always.

Reply:
Actually, you already placed that code but commented out. Why is that?

Reply:
I can use the following way:
```C#
	if (Selector is ListView lv)
	{
		ApplyMultiSelectState(lv.IsSelectionMultiple);
	}
	else if (Selector is GridView gv)
	{
		ApplyMultiSelectState(gv.IsSelectionMultiple);
	}
```
But this violates the "Open-Closed principle".
With this implementation, I will have to change **OnLoaded** method every time if I have new elements for the multi-select function, or new functionality.
Consequently, I have a higher chance of getting a conflict when merging into a master/main branch.

A more correct solution would be to place this functionality for each SelectorItem type.

Reply:
Thanks! This indeed makes sense. We can certainly extract the `ListView` specific code as well as the one you changed for `GridView` in the `OnLoaded` in the same manner.

Reply:
Done